spec/controllers/faqs_controller_spec.rb

```diff
@@ -0,0 +1,166 @@
+require 'rails_helper'
+
+RSpec.describe FaqsController, type: :controller do
+  let(:my_faq) { Faq.create!( question: "What is the answer to life the universe and everything?", answer: "42") }
+  let(:admin) { User.create!(name: 'admin user', email: 'admin@admin.com', password: "helloworld", role: :admin) }
+
+  context "guest" do
+    describe "GET index" do
+      it "returns http success" do
+        get :index
+        expect(response).to have_http_status(:success)
+      end
+
+      it "assigns Faqs.all to faq" do
+        get :index
+        expect(assigns(:faqs)).to eq([my_faq])
+      end
+    end
+
+    describe "GET new" do
+      it "returns http redirect" do
+        get :new
+        expect(response).to redirect_to(new_session_path)
+      end
+    end
+
+    describe "FAQS create" do
+      it "returns http redirect" do
+        post :create, faq: { question: "What is the answer to life the universe and everything?", answer: "42" }
+        expect(response).to redirect_to(new_session_path)
+      end
+    end
+
+    describe "GET edit" do
+      it "returns http redirect" do
+        get :edit, id: my_faq.id
+        expect(response).to redirect_to(new_session_path)
+      end
+    end
+
+    describe "PUT update" do
+      it "returns http redirect" do
+        new_question = "What is the question?"
+        new_answer = "I'll need some time to think about that"
+
+        put :update, id: my_faq.id, faq: {question: new_question, answer: new_answer}
+        expect(response).to redirect_to(new_session_path)
+      end
+    end
+
+    describe "DELETE destroy" do
+      it "returns http redirect" do
+        delete :destroy, id: my_faq.id
+        expect(response).to redirect_to(new_session_path)
+      end
+    end
+  end
+
+  context "admin" do
```

Comment:
you need to have a non-admin test section as well (if it's below, then I am sorry)